### PR TITLE
Feat/ethers disconnect

### PR DESCRIPTION
**Pull request by svenvoskamp**
* Add  function `disconnect` to exports

Closes https://github.com/WalletConnect/web3modal/issues/1439

### CODE REVIEW

File: packages/ethers5/exports/react.tsx [hunk 1]
```diff
@@ -46,11 +48,26 @@ export function useWeb3ModalAccount() {
...
   const chainId = state.chainId
+  const provider = state.provider
+  const providerType = state.providerType
+
+  async function disconnect() {
+    localStorage.removeItem(WALLET_ID)
+    ProviderController.reset()
+
+    if (providerType === 'injected' || providerType === 'eip6963') {
+      provider?.emit('disconnect')
```
[the remark this answers is not in the document]
Comment (svenvoskamp) [reply packages/ethers5/exports/react.tsx:59]: This triggers `disconnectHandler` inside the client, which clears `localStorage`

File: packages/ethers5/src/client.ts [hunk 2]
```diff
@@ -313,6 +313,20 @@ export class Web3Modal extends Web3ModalScaffold {
     return ProviderController.subscribe(callback)
   }
 
+  public async disconnect() {
+    const provider = ProviderController.state.provider
```
Comment (0xAsimetriq) [inline packages/ethers5/src/client.ts:317]:
```suggestion
    const {provider, providerType} = ProviderController.state
```

File: packages/ethers5/exports/react.tsx [hunk 3]
```diff
@@ -46,11 +48,26 @@ export function useWeb3ModalAccount() {
   const address = state.address
   const isConnected = state.isConnected
   const chainId = state.chainId
+  const provider = state.provider
+  const providerType = state.providerType
+
+  async function disconnect() {
```
Comment (0xAsimetriq) [inline packages/ethers5/exports/react.tsx:54]: Let's keep consistent and do `useDisconnect` that returns disconnect function. Also, it seems like same logic as in `client` is replicated there. Can we not just use `ConnectionController.disconnect` at this stage (same as all other hooks use controllers for things like theme etc?)

**Review by 0xAsimetriq — CHANGES_REQUESTED** [review]

**Review by 0xAsimetriq — APPROVED** [review]
Much cleaner 👌 ty for changes

**Review by glitch-txs — APPROVED** [review]

**Comment by GorgeousPuree** [comment]
Please update documentation with this new hook.
https://docs.walletconnect.com/web3modal/react/about

**Pull request merged**